lib/modules/apostrophe-schemas/public/js/user.js

```diff
@@ -1040,6 +1039,41 @@ apos.define('apostrophe-schemas', {
       convert: convertString
     });
 
+    self.addFieldType({
+      name: 'colorpicker',
+      populate: function (data, name, $field, $el, field, callback) {
+        $field.val(data[name]);
+        apos.ui.enhanceColorpicker($field);
+        $field.next().text(data[name]);
+        $field.on('change', function() {
+          $field.next().text($field.val());
+        });
+        return setImmediate(callback);
+      },
+      convert: function (data, name, $field, $el, field, callback) {
+        data[name] = $field.val();
+        return setImmediate(callback);
+      }
+    });
+
+    self.addFieldType({
+      name: 'range',
+      populate: function (data, name, $field, $el, field, callback) {
+        $field.val(data[name]);
+        var $valueIndicator = $field.prev();
+        $valueIndicator.text($field.val());
+        $field.on('input', function() {
+          $valueIndicator.text($field.val());
+          $field.trigger('change');
+        });
+        return setImmediate(callback);
+      },
+      convert: function (data, name, $field, $el, field, callback) {
+        data[name] = $field.val();
```

Comment:
You should reject bad input here too, so users don't have to wait until they save a doc with three levels of nested arrays to find out it's mysteriously discarded by the server-side converter (which does have to start doing that, as noted earlier).